R/SoilProfileCollection-operators.R

```diff
@@ -122,27 +122,64 @@ setMethod("[", signature(x = "SoilProfileCollection",
 
             # subset horizons/slices based on j --> only when j is given
             if (!missing(j)) {
+
               # faster replacement of j subsetting of horizon data
-              j.res <- as.list(aggregate(
-                h[[hzidname(x)]],
-                by = list(h[[idname(x)]]),
-                FUN = function(hh) {
-                  list(1:length(hh) %in% j)
-                },
-                drop = FALSE
-              )$x)
-              j.idx <- which(do.call('c', j.res))
-              h <- h[j.idx, ]
+              if(aqp_df_class(x) == "data.table") {
+
+                # data.table can do this much more efficiently
+                if (requireNamespace("data.table", quietly = TRUE)) {
+
+                  bylist <- list(h[[idname(x)]])
+                  names(bylist) <- idname(x)
+
+                  .idx.in.range <- function(len, jj) {
+                    1:len %in% jj
+                  }
+
+                  # this one was a head scratcher but is super cool! and pretty fast
+                  #  the main overhead comes from [,data.table ...
+                  # if(!data.table::is.data.table(h))
+                  #   h <- as.data.table(h)
+
+                  h <- h[, `:=`(.jmatch = .idx.in.range(.N, j),
```

Comment:
Whoa, this is new to me. I trust that this is well tested.

Reply:
I guess that means you didn't run the "big random" demo I tagged a couple days ago. Try it.

Reply:
I reviewed it, and running something doesn't confer understanding of the implementation. I still have not had time to review much of the `data.table` functionality, other  than what is related to the stuff I used in my re-write of `slice`. My comment had to do with the syntax: I have never seen `.idx.in.range` used in this context.

Reply:
.idx.in.range is an internal new function that is defined right above that call. So I don't know where you would have seen it other than in my code.